src/proxy-ui-api/src/test/java/org/niis/xroad/restapi/service/TokenCertificateServiceTest.java

```diff
@@ -79,58 +119,359 @@
     @MockBean
     private CertificateAuthorityService certificateAuthorityService;
 
-    @MockBean
+    @SpyBean
     private KeyService keyService;
 
+    @MockBean
+    private GlobalConfService globalConfService;
+
+    @MockBean
+    private GlobalConfFacade globalConfFacade;
+
+    @MockBean
+    private ClientRepository clientRepository;
```

Comment:
These are probably redundant here

Reply:
I guess they are not? I mean if I don't mock those, TokenCertificateService will e.g. use real GlobalConfService, which we do not want here since it's not meant to be an integration test.

This is pretty clumsy way to do it though, I agree, and usually you forget to add some dependencies as mocks, and can easily call the real implementation by mistake. I wonder if there's a better way to do this, or if it makes sense to use this approach at all.

Reply:
Thought a bit more about and realized that if we mock everything, TokenCertificateServiceTest is then a pure unit test, so what sense does it make to implement it as a (slow) SpringBootTest? Would it not be better to just have it as a regular junit test which is more lightweight and guarantees we don't accidentally use some injected real dependencies?

And then a counterpoint, spring test framework makes constructing, injecting and mocking the dependencies easier (probably, not sure how it would be done in pure mockito) and some of the tests use spring features such as mock users.

I have a feeling this is related to the problem with slow tests, I need to ask for help somewhere. I don't think it should stop this PR though.

Reply:
In this case I couldn't find any usage (direct or indirect) for the fields in question. But it might be good to mock them anyway to ensure nothing calls the real services.

It would be very nice to get the tests to execute faster but yeah that is for another task